### PR TITLE
add vmgateway helm chart

### PR DESCRIPTION
**Pull request by tenmozes**

### CODE REVIEW

File: charts/victoria-metrics-gateway/templates/ingress.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+{{- if .Values.ingress.enabled }}
+{{- $ingressApiIsStable := eq (include "vmauth.ingress.isStable" .) "true" -}}
```
Comment (f41gh7) [inline charts/victoria-metrics-gateway/templates/ingress.yaml:2]: Should it be vmauth -> vmwagateway?

**Review by f41gh7 — APPROVED** [review]
LGTM

File: charts/victoria-metrics-gateway/templates/ingress.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+{{- $ingressSupportsIngressClassName := eq (include "vmauth.ingress.supportsIngressClassName" .) "true" -}}
+{{- $ingressSupportsPathType := eq (include "vmauth.ingress.supportsPathType" .) "true" -}}
```
Comment (k1rk) [inline charts/victoria-metrics-gateway/templates/ingress.yaml:4]: this templates are missing from `_helpers.tpl`

also as @f41gh7 mentioned we should also use unique name like `vmgtw.ingres..`

File: charts/victoria-metrics-gateway/templates/service-monitor.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
+{{- if .Values.serviceMonitor.enabled -}}
+apiVersion: monitoring.coreos.com/v1
```
Comment (k1rk) [inline charts/victoria-metrics-gateway/templates/service-monitor.yaml:2]: should we use native VM resource here? i mean service scrape?

Comment (tenmozes) [reply charts/victoria-metrics-gateway/templates/service-monitor.yaml:2]: just to be compatible with other charts

**Review by f41gh7 — APPROVED** [review]
LGTM

**Comment by f41gh7** [comment]
Thanks!

**Pull request merged**